src/store/greetingSlice.js

```diff
@@ -0,0 +1,35 @@
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
+
+const initialState = {
+  greeting: '',
+  isLoading: false,
+  error: undefined,
+};
+
+const fetchGreeting = createAsyncThunk('greeting/fetchGreeting', async () => {
+  const response = await fetch('http://localhost:3001/api/v1/greetings');
+  const data = await response.json();
+  return data.message;
```

Comment:
- [ ] I noticed you used 3001 instead of 3000 since this is the api and the rails app would run first and the React app would run on 3001.
- [ ] Also, I could not see the data rendefr on the React frontend and it is because of CORS issue as it is in the image below. You can read more about cors and how to fix it [here](https://medium.com/@chris.lty07/configuring-for-cors-and-cookies-with-react-and-ruby-on-rails-1c30219cc2db).
![image](https://github.com/wacoo/hello-react-front-end/assets/86200356/589335c3-97ec-47f2-b63a-43ce9314529a)